tests/foreman/performance/common/perf_candlepin.py

```diff
@@ -0,0 +1,94 @@
+    '''
+    @classmethod
+    def single_register_activation_key(cls, ak_name, default_org, vm):
+        # note: must create ssh keys for vm if running on local
+        # result = ssh.command('subscription-manager clean', hostname=vm)
+        result = ssh.command('subscription-manager clean', hostname=vm)
+        if result.return_code != 0:
+           cls.logger.error('Failed to clean sub-mgr!')
+           cls.logger.error(result.stderr)
+
+        result = ssh.command('time -p subscription-manager register --activationkey={0} --org={1}'.format(ak_name, default_org), hostname=vm)
+        # if result.return_code == 0:
+        real_time = [real for real in result.stderr.split('\n') if real.startswith('real')]   
+        real_time = real_time[0].split(' ')[1]
+        return float(real_time)
+
+    '''
+    Subscribe VM to Satellite by Register + Attach
+    '''
+    @classmethod
+    def single_register_attach(cls, sub_id, default_org, environment, vm):
+        result = ssh.command('subscription-manager clean', hostname=vm)
+        time_reg = cls.sub_mgr_register_authentication(default_org, environment, vm)
+        # pool_id = cls.get_subscription_pool_id()
+        time_att = cls.sub_mgr_attach(sub_id, vm)
+        
+        if time_reg == None or time_att == None:
+            total_time = 0
+        else:
+            total_time = time_reg+time_att
+        return time_reg, time_att, total_time
+
+    '''
+    Helper Function: subscription-manager register -u -p --org --environment
+    '''
+    @classmethod
+    def sub_mgr_register_authentication(cls, default_org, environment, vm):
+        # result = ssh.command('subscription-manager clean', hostname=cls.vm)
+        result = ssh.command('time -p subscription-manager register --username={0} --password={1} --org={2} --environment={3}'.format(cls.username, cls.password, default_org, environment), hostname=vm)
+        if result.return_code == 0:
+           real_time = [real for real in result.stderr.split('\n') if real.startswith('real')]
+           real_time = real_time[0].split(' ')[1]
+           return float(real_time)
+
+    '''
+    Helper Function: subscription-manager attach --pool=pool_id
+    '''
+    @classmethod
+    def sub_mgr_attach(cls, pool_id, vm):
+        result = ssh.command('time -p subscription-manager attach --pool={}'.format(pool_id), hostname=vm)
+        if result.return_code == 0:
+           real_time = [real for real in result.stderr.split('\n') if real.startswith('real')]
+           real_time = real_time[0].split(' ')[1]
+           return float(real_time)
+
+    '''
+    Delete system from subscription
+    '''
+    @classmethod
+    def single_delete(cls, uuid):
+        # delete by each uuid and return time result
+        start = time.time()
+        r = requests.delete('https://{0}/katello/api/systems/{1}'.format(cls.serverhost, uuid), verify=False, auth=(cls.username,cls.password))
+        if r.status_code == 204:
```

Comment:
Rename the variable `r` to something more meaningful, like `response`. Also, use names, not bare status codes:

```python
response = requests.delete(…)
if response.status_code == httplib.NO_CONTENT:
    …
```